sklearn/inspection/_plot/partial_dependence.py

```diff
@@ -1422,26 +1423,28 @@ def plot(
                     default_ice_lines_kws = {}
                     default_pd_lines_kws = {}
 
+                line_kw = _validate_style_kwargs(default_line_kws, line_kw)
+                ice_lines_kw = _validate_style_kwargs(
+                    default_ice_lines_kws, ice_lines_kw
+                )
+                pd_line_kw = _validate_style_kwargs(default_pd_lines_kws, pd_line_kw)
+
                 ice_lines_kw = {
-                    **default_line_kws,
-                    **default_ice_lines_kws,
                     **line_kw,
                     **ice_lines_kw,
                 }
                 del ice_lines_kw["label"]
 
                 pd_line_kw = {
-                    **default_line_kws,
-                    **default_pd_lines_kws,
                     **line_kw,
                     **pd_line_kw,
                 }
```

Comment:
Yep here we have to handle that the `pd_*` and `ice_*` take over the `line_*`.

The following should work:

```
                default_ice_lines_kws = {**default_line_kws, **default_ice_lines_kws}
                default_pd_lines_kws = {**default_line_kws, **default_pd_lines_kws}

                line_kw = _validate_style_kwargs(default_line_kws, line_kw)

                ice_lines_kw = _validate_style_kwargs(
                    _validate_style_kwargs(default_ice_lines_kws, line_kw),
                    ice_lines_kw
                )
                del ice_lines_kw["label"]

                pd_line_kw = _validate_style_kwargs(
                    _validate_style_kwargs(default_pd_lines_kws, line_kw),
                    pd_line_kw
                )
```

I'll make a push